src/routes/docs/tutorials/pagination/step-6/+page.markdoc

```diff
@@ -0,0 +1,104 @@
+---
+layout: tutorial
+title: Adding tasks
+description: Add tasks to your React application using Appwrite
+step: 6
+---
+# Add Tasks context {% #add-todos-page %}
+
+Now we need to display our Todos to the page. In React, we can use [context](https://reactjs.org/docs/context.html) to share data between components. We'll use context and create a simple custom hook to manage out tasks.
+
+Create a new file `src/lib/context/tasks.jsx` and add the following code to it.
+
+```js
+import { createContext, useContext, useEffect, useState } from "react";
+import { databases } from "../appwrite";
+import { Query } from "appwrite";
+
+export const TODOS_DATABASE_ID = "[YOUR_DATABASE_ID]";
+export const TODOS_COLLECTION_ID = "[YOUR_COLLECTION_ID]";
+
+const TasksContext = createContext();
+
+export function useTasks() {
+  return useContext(TasksContext);
+}
+
+export function TasksProvider(props) {
+  const [tasks, setTasks] = useState([]);
+
+  async function init() {
+    try {
+      const response = await databases.listDocuments(
+        TODOS_DATABASE_ID,
+        TODOS_COLLECTION_ID,
+        [Query.orderDesc("$createdAt"), Query.limit(4)],
+      );
+      setTasks(response.documents);
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  useEffect(() => {
+    init();
+  }, []);
+
+  return (
+    <TasksContext.Provider value={{ current: tasks }}>
+      {props.children}
+    </TasksContext.Provider>
+  );
+}
+```
+we will add more functionality to the context later
+
+# Basic Routing
+
+First, wrap the `main` element with the `TasksProvider` component.
+
+Update `src/App.jsx` to the following code.
+
+```jsx
+import { TasksProvider } from "./lib/context/tasks";
+import { Tasks } from "./pages/Tasks";
+
+function App() {
+  return (
+    <div>
+      <TasksProvider>
+        <Tasks />
+      </TasksProvider>
+    </div>
+  );
+}
+
+export default App;
+```
+
+# Tasks page
```

Comment:
Some of these headings are missing {% task-page %} tags that helps index our table of context!

Reply:
is there a way to format the headings with missing tags into headings that have a lower level? H1>H2>H3...

Reply:
# Tasks page {% #task-page %}
## Tasks page {% #task-page %}
### Tasks page {% #task-page %}

Reply:
ok, gotcha